packages/components/src/core/MenuItem/index.stories.tsx

```diff
@@ -61,6 +61,316 @@ export const Default = {
+        })}
+      </div>
+    );
+  }
+
+  // loop through all DISABLED_OPTIONS + create headers for SELECTED_OPTIONS
+  function MenuItemSelected({
+    isMultiSelect,
+    details,
+    column,
+    sdsIcon,
+    selected,
+  }: {
+    isMultiSelect: (typeof MULTI_SELECT_OPTIONS)[number];
+    details: (typeof DETAILS_OPTIONS)[number];
+    column: (typeof COLUMN_OPTIONS)[number];
+    sdsIcon: (typeof ICON_OPTIONS)[number];
+    selected: (typeof SELECTED_OPTIONS)[number];
+  }) {
+    const LABEL_STYLE: React.CSSProperties = {
+      ...MID_LABEL,
+      alignSelf: "end",
+      borderWidth: "1px",
+      fontSize: "0.83em",
+      margin: "0 0 10px 0",
+    };
+    return (
+      <div style={DISPLAY_CONTENTS}>
+        <p style={LABEL_STYLE}>
+          Selected: <b>{selected ? "true" : "false"}</b>
+        </p>
+        {DISABLED_OPTIONS.map((disabled) => {
+          return (
+            <MenuItemDisabled
+              isMultiSelect={isMultiSelect}
+              details={details}
+              column={column}
+              sdsIcon={sdsIcon}
+              selected={selected}
+              disabled={disabled}
+              key={String(disabled)}
+            />
+          );
+        })}
+      </div>
+    );
+  }
+
+  // loop through all PSEUDO_STATES + create headers for DISABLED_OPTIONS, PSEUDO_STATES
+  function MenuItemDisabled({
+    isMultiSelect,
+    details,
+    column,
+    sdsIcon,
+    selected,
+    disabled,
+  }: {
+    isMultiSelect: (typeof MULTI_SELECT_OPTIONS)[number];
+    details: (typeof DETAILS_OPTIONS)[number];
+    column: (typeof COLUMN_OPTIONS)[number];
+    sdsIcon: (typeof ICON_OPTIONS)[number];
+    selected: (typeof SELECTED_OPTIONS)[number];
+    disabled: (typeof DISABLED_OPTIONS)[number];
+  }) {
+    const LEVEL_STYLE: React.CSSProperties = {
+      marginBottom: 20,
+    };
+    const LABEL_STYLE: React.CSSProperties = {
+      fontSize: "0.67em",
+      margin: "10px 0 20px 0",
+    };
+    return (
+      <div style={DISPLAY_CONTENTS}>
+        {PSEUDO_STATES.map((state) => {
+          return (
+            // to pass a11y, this container div must have role of "menu" since it contains components with roles of "menuitem"
+            <div role="menu" style={LEVEL_STYLE}>
+              {/* removes irrelevant disabled iterations: when combined with all pseudo-states except default, `disabled=false` is impossible */}
+              {(disabled === false ||
+                (disabled === true && state === "default")) && (
+                <>
+                  <p style={LABEL_STYLE}>
+                    {disabled ? "Disabled: " : "State: "}
+                    <b>{disabled ? "true" : state}</b>
+                  </p>
+                  <RawMenuItem
+                    {...props}
+                    data-testid="menu-item"
+                    isMultiSelect={isMultiSelect}
+                    details={details}
+                    column={column}
+                    sdsIcon={sdsIcon}
```

Comment:
The `sdsIcon` prop is not recognizing the type I assigned it, I think because icons used within MenuItem are already assigned their own type (`IconName`) in the index.tsx file. But I cannot figure out how to transport that "type" into the Screenshot Test story... I tried similar approaches to what Timmy helped me with, in the ButtonIcon Screenshot Test (not yet merged in, so will have to [look at the PR for it](https://github.com/chanzuckerberg/sci-components/pull/462/files#r1192814051)), but I don't know TypeScript yet, so I just don't know what is going wrong with my attempts...

Reply:
this should be resolved by my latest commit! Thank you!